File: backend/scripts/fetch-algeria.js
const https = require('https');
const fs = require('fs');
const path = require('path');

function fetchJson(url) {
  return new Promise((resolve, reject) => {
    https.get(url, (res) => {
      let data = '';
      res.on('data', chunk => data += chunk);
      res.on('end', () => resolve(JSON.parse(data)));
    }).on('error', reject);
  });
}

async function run() {
  try {
    console.log('Fetching Wilayas...');
    const wilayas = await fetchJson('https://raw.githubusercontent.com/BrahimS/algeria-api/master/data/wilayas.json');
    console.log('Fetching Communes...');
    const communes = await fetchJson('https://raw.githubusercontent.com/BrahimS/algeria-api/master/data/communes.json');

    const algeriaData = {};
    wilayas.forEach(w => {
      algeriaData[w.id] = {
        name: w.name,
        communes: []
      };
    });

    communes.forEach(c => {
      if (algeriaData[c.wilaya_id]) {
        algeriaData[c.wilaya_id].communes.push(c.name);
      }
    });

    const jsContent = `const algeriaData = ${JSON.stringify(algeriaData, null, 2)};`;
    const outputPath = path.join(__dirname, '../frontend/js/algeria-cities.js');
    fs.writeFileSync(outputPath, jsContent);
    console.log(`Success! Wrote ${Object.keys(algeriaData).length} wilayas to ${outputPath}`);
  } catch (error) {
    console.error('Error:', error);
  }
}

run();

File: backend/scripts/fetch-algeria.py
import urllib.request
import json
import os

wilayas_url = 'https://raw.githubusercontent.com/AbderrahmeneDZ/Wilaya-Of-Algeria/master/Wilaya_Of_Algeria.json'
communes_url = 'https://raw.githubusercontent.com/AbderrahmeneDZ/Wilaya-Of-Algeria/master/Commune_Of_Algeria.json'

print("Fetching Wilayas...")
with urllib.request.urlopen(wilayas_url) as response:
    wilayas = json.loads(response.read().decode())

print("Fetching Communes...")
with urllib.request.urlopen(communes_url) as response:
    communes = json.loads(response.read().decode())

algeriaData = {}

for w in wilayas:
    # w might have "id", "name"
    # let's map by id
    w_id = str(w['id'])
    algeriaData[w_id] = {
        'name': w['name'],
        'communes': []
    }

for c in communes:
    w_id = str(c['wilaya_id'])
    if w_id in algeriaData:
        algeriaData[w_id]['communes'].append(c['name'])

out_path = os.path.join(os.path.dirname(__file__), '../../frontend/js/algeria-cities.js')
with open(out_path, 'w', encoding='utf-8') as f:
    f.write('const algeriaData = ' + json.dumps(algeriaData, indent=2, ensure_ascii=False) + ';')

print(f"Success! Wrote {len(algeriaData)} wilayas to {out_path}")
